src/YagrCore/index.ts

```diff
@@ -225,6 +225,27 @@ class Yagr<TConfig extends MinimalValidConfig = MinimalValidConfig> {
         return this.uplot.series[this.state.y2uIdx[id]];
     }
 
+    getSerieLegendColor(serie: Series) {
+        const {legendColorKey, color, lineColor} = serie;
+
+        let serieColor = color;
+
+        switch (legendColorKey) {
+            case 'lineColor': {
+                if (lineColor) {
+                    serieColor = lineColor;
+                }
+                break;
+            }
+            case 'color':
+            default: {
```

Comment:
There is no need in default case, or u don't need this line of code
```js
let serieColor = color;
```

Reply:
Removed default case